src/sentry/snuba/metrics/extraction.py

```diff
@@ -905,7 +905,30 @@ def _deep_sorted(value: Union[Any, Dict[Any, Any]]) -> Union[Any, Dict[Any, Any]
 
 
 def are_specs_equal(spec_1: MetricSpec, spec_2: MetricSpec) -> bool:
-    return _deep_sorted(spec_1) == _deep_sorted(spec_2)
+    equal = True
+    if spec_1.keys() != spec_2.keys():
+        equal = False
+
+    if equal:
+        for key, value in spec_1.items():
+            if key == "tags":
+                return _compare_lists(spec_1["tags"], spec_2["tags"])
+
+            elif spec_2.get(key) != value:
+                equal = False
```

Comment:
I don't know if iterating spec_1 actually ensures equality in the odd case that spec 2 has a new key. We should also check # of keys is equal.

Reply:
We do that on line 909:
> if spec_1.keys() != spec_2.keys():
>     equal = False